### PR TITLE
Integrate new colors

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
All of us

## What is it doing for them?
Integrates the new colour palette.
Updates the Palette story, the goal is to add more details as to what the goal is for each colour section (core, accent, neutrals, warm neutrals, validation).

## Notes and next steps
The codebase currently uses the `newColors` object, the `NewPaletteColors` type and the `newColors()` function. This is to help ensure we've covered all grounds. I'm concerned in the time it take to review and merge, conflicting PRs will cause us to forget some newer changes and this helps catch them.
Once it's merged in, I'll get a new PR up to revert to `colors`, `PaletteColors` and `colors()` and hopefully get that out very quickly.

Once it goes to staging, it would be good to have a short freeze on prod deployments just so we can make sure a11y and our e2e still passes. I've tested as much as I could every step of the way so I'm hoping it works out.

### CODE REVIEW

File: common/views/themes/config.ts
Comment (rcantin-w) [inline common/views/themes/config.ts:326]: yeah this makes me unhappy, I'll have to investigate how this could be done otherwise.

File: cardigan/stories/global/typography/typography.stories.tsx
Comment (rcantin-w) [inline cardigan/stories/global/typography/typography.stories.tsx:15]: Enforcing validation.red only being used for validation

File: common/views/components/Footer/FooterSocial.tsx
Comment (rcantin-w) [inline common/views/components/Footer/FooterSocial.tsx:127]: Did this as discussed the Icons being green with Gareth and that's what's sparked the eventual refactor of the Footer we're going to do; eventually icons will be set to `currentColor` by default as well.

File: content/webapp/types/color-selections.ts
Comment (rcantin-w) [inline content/webapp/types/color-selections.ts:3]: Should we move away from red and green here?

File: common/views/components/ButtonSolid/ButtonSolid.tsx
Comment (rcantin-w) [inline common/views/components/ButtonSolid/ButtonSolid.tsx:176]: Aligning behaviour of buttons until we decide how we want to do hover behaviour. Gareth O was suggesting lightening the colours, we talked about using a function, etc. We'll need to decide how that looks on outline buttons as well.

**Comment by alexwlchan** [comment]
I’ll dive into this after the retro.

File: common/views/themes/config.ts
Comment (rcantin-w) [inline common/views/themes/config.ts:159]: https://github.com/wellcomecollection/wellcomecollection.org/issues/8542 Naming issues outlined here

File: common/views/components/Dot/Dot.tsx
Comment (rcantin-w) [inline common/views/components/Dot/Dot.tsx:20]: Added checks in `classNames` to transform `accent.green` to `accent-green` as dots in class names are no nos.

File: common/views/themes/base/container.ts
Comment (rcantin-w) [inline common/views/themes/base/container.ts:69]: Was only used once (`content/webapp/pages/stories.tsx`) so I moved it there directly.

File: content/webapp/services/prismic/types/series.ts
Comment (rcantin-w) [inline content/webapp/services/prismic/types/series.ts:23]: See comments about Series colours https://github.com/wellcomecollection/wellcomecollection.org/pull/8506/files#r980106445

File: common/views/components/Dot/Dot.tsx
Comment (alexwlchan) [inline common/views/components/Dot/Dot.tsx:20]: Why do we need `classNames()` here?

File: common/views/components/PopupDialog/PopupDialog.tsx
Comment (alexwlchan) [inline common/views/components/PopupDialog/PopupDialog.tsx:150]: Why have you removed this line?

File: content/webapp/components/HTMLSerializers/HTMLSerializers.tsx
Comment (alexwlchan) [inline content/webapp/components/HTMLSerializers/HTMLSerializers.tsx:176]: Not necessarily your bug, but why do we have two spans here? 🤔

File: common/views/components/Dot/Dot.tsx
Comment (rcantin-w) [reply common/views/components/Dot/Dot.tsx:20]: I left a comment just above, I went through the doc and explained as much as I could for unclear things

File: cardigan/stories/components/TabNav/TabNav.stories.tsx
Comment (rcantin-w) [reply cardigan/stories/components/TabNav/TabNav.stories.tsx:76]: They aren't used anymore and I hadn't cleaned it up in a prior merge

File: cardigan/stories/components/TabNav/TabNav.stories.tsx
Comment (rcantin-w) [reply cardigan/stories/components/TabNav/TabNav.stories.tsx:54]: Because they were missing, TabNavV2 requires them and they hadn't been added in a prior merge

File: common/views/components/PopupDialog/PopupDialog.tsx
Comment (rcantin-w) [reply common/views/components/PopupDialog/PopupDialog.tsx:150]: I'm really unsure, I wonder if it's a merge conflict leftover 🤔 I'll dig.

File: cardigan/stories/global/palette/palette.stories.tsx
Comment (alexwlchan) [inline cardigan/stories/global/palette/palette.stories.tsx:150]: Having these descriptions are really nice for understanding why the colours are used. ✨

File: common/views/components/Dot/Dot.tsx
Comment (rcantin-w) [reply common/views/components/Dot/Dot.tsx:20]: It's not ideal but one day we'll get rid of font-[color] 👀

File: common/utils/classnames.ts
Comment (alexwlchan) [inline common/utils/classnames.ts:50]: nit: let's link to the ticket here https://github.com/wellcomecollection/wellcomecollection.org/issues/8418

File: content/webapp/components/HTMLSerializers/HTMLSerializers.tsx
Comment (rcantin-w) [reply content/webapp/components/HTMLSerializers/HTMLSerializers.tsx:176]: Good catch, I'll have a looksy

File: content/webapp/pages/stories.tsx
Comment (alexwlchan) [inline content/webapp/pages/stories.tsx:77]: styyyyyyles :D

**Review by alexwlchan — APPROVED** [review]
🌈 🖌️ 

A couple of small issues inline (sorry if I came across as a little negative/naggy in the first batch of comments), but this looks great for a PR of this size.

I move to get this merged ASAP and work on issues as we find them. 🦑

File: cardigan/stories/global/palette/palette.stories.tsx
Comment (rcantin-w) [reply cardigan/stories/global/palette/palette.stories.tsx:150]: Based off of Gareth's official ones, may his Wellcome spirit live on in our codebase 😄

File: common/views/components/PopupDialog/PopupDialog.tsx
Comment (rcantin-w) [reply common/views/components/PopupDialog/PopupDialog.tsx:150]: No idea why and actually thanks, this showed me that on the line just after, the background color was wrong ! (set to `purple` instead of `accent.purple`). Guess I changed the wrong something in that file.

**Comment by rcantin-w** [comment]
> I move to get this merged ASAP and work on issues as we find them. 🦑

Sounds good to me! I've pushed some changes based on your comments, thanks for looking so quickly! There shouldn't be issues with tests  🤞 I'll merge it first thing tomorrow after a final merge from `main` and a double check.

File: content/webapp/pages/stories.tsx
Comment (rcantin-w) [inline content/webapp/pages/stories.tsx:90]: Adding this to https://github.com/wellcomecollection/wellcomecollection.org/issues/8540

File: cardigan/stories/global/palette/palette.stories.tsx
Comment (davidpmccormick) [inline cardigan/stories/global/palette/palette.stories.tsx:19]: Shouldn't these `string`s be `number`s?

**Review by davidpmccormick — COMMENTED** [review]
Can't believe I made it to the end 🤯 . Nice one!

I guess I must be wrong about the `rgb`/`hsl` typing otherwise TypeScript would be shouting, but I couldn't immediately see why.

**Comment by alexwlchan** [comment]
> I guess I must be wrong about the rgb/hsl typing otherwise TypeScript would be shouting, but I couldn't immediately see why.

Good spot! I think if you dig in, the types are so heavily nested that at some point it `any`s them and the type checker gets stuck.

File: cardigan/stories/global/palette/palette.stories.tsx
Comment (rcantin-w) [reply cardigan/stories/global/palette/palette.stories.tsx:19]: That is super weird, I've done some tests and it really doesn't seem to care. I'll switch them, but it seems a bit useless. What @alexwlchan says might make sense... If I type them separately and tell the function what to expect as an output, then it checks, but I'm thinking it might be overkill in this case?

Comment (alexwlchan) [reply cardigan/stories/global/palette/palette.stories.tsx:19]: I think the issue is the massive object with all the unpacking; if you break it up and add subtypes, the issue falls out.

Comment (rcantin-w) [reply cardigan/stories/global/palette/palette.stories.tsx:19]: Yeah that's what I meant by typing separately - subtypes! Just doing that doesn't flag anything either though, I think it needs that AND adding an expected output (which I guess we should've had anyway!), only then does it start flagging errors for me.
So like:
`function hexToRgb(hex): RGB {[...]}`

Feels like a lot for this, but otherwise our typing isn't useful, so I'll make a PR.

That was an interesting one, thanks @davidpmccormick !

**Pull request merged**
